base/math.jl

```diff
@@ -567,7 +567,7 @@ end
 
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
-hypot(x::Number...) = sqrt(sum(abs2(y) for y in x))
+hypot(x::Number...) = reduce(hypot, x)
```

Comment:
This will give the wrong results for `hypot(-2)`.   It may be simplest to define the special case `hypot(x::Number) = abs(x)`.

Reply:
Added `hypot(x::Number) = abs(x)`.

Reply:
Although I'm not 100% sure if `hypot` with a single argument makes a lot of sense.